__mocks__/react-dom.js

```diff
@@ -0,0 +1,14 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const reactDOM = jest.genMockFromModule('fs'); // eslint-disable-line no-undef
+
+// Overwrite this in the test to mock it.
+reactDOM.__findDOMNode = () => {};
+
+reactDOM.findDOMNode = function(...args) {
+  return reactDOM.__findDOMNode(...args);
+};
+
+module.exports = reactDOM;
```

Comment:
I see various issues in this mock, but especially I think it's a lot simpler to do it like I did here: https://github.com/devtools-html/perf.html/blob/master/src/test/components/ProfileThreadTracingMarkerOverview.test.js#L16-L18

The issues I see for posterity:
* why using the module `'fs'` as template ?
* instead of disabling the rule `no-undef` [you can declare which globals are expected](http://eslint.org/docs/user-guide/configuring#specifying-globals): `/* global jest */`